contrib/packs/actions/check_pack_auto_deploy.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import requests
+import json
+import getopt
+import argparse
+import os
+import yaml
+
+from getpass import getpass
+from st2actions.runners.pythonrunner import Action
+
+class Packs_Check_Pack_Auto_Deploy(Action):
+    def __init__(self, config=None):
+        super(Packs_Check_Pack_Auto_Deploy, self).__init__(config=config)
+
+    def run(self, branch, repository):
+        exit_code = 1
+
+        # Set up the results
+        results = {}
+        results['notify_channel'] = None
+        results['deployment_branch'] = None
+
+        if self.config.has_key("repositories"):
```

Comment:
Minor formatting thing - to avoid that many levels of nesting and make the code a bit easier to follow, you could do something like that:

```python
repositories = self.config.get('repositories', {})
repository = repositories.get(repository, {})
auto_deploy = repository.get('auto_deployment, False)

if auto_deploy:
....
```